pilot/pkg/xds/delta_test.go

```diff
@@ -329,6 +330,10 @@ func TestDeltaReconnectRequests(t *testing.T) {
 	}
 }
 
+func init() {
+	features.EnableAmbient = true
```

Comment:
Is there a reason we want to enable ambient for all tests in the xDS package now? Can we not use `SetforTest`?

Reply:
I think it was due to a race due to some components not having a proper "synchronously wait for all asynchronous tasks to complete". I'll revert this and double check. Not sure why this PR would trigger that tbh so maybe it's not needed anymore